# Basic Python

1. Numbers

In [1]:
2 + 2

4

In [2]:
60 - 2 * 4

52

In [3]:
60 - (2 * 4)/8

59.0

In [4]:
6/8

0.75

In [5]:
17 / 3

5.666666666666667

In [6]:
17 // 3

5

/ --> gives the floating point division
// --> gives floor division

In [8]:
17 % 3

2

In [9]:
6 ** 2

36

In [10]:
2 * 6.24 + 1

13.48

Float cuz one of the number is in float

3. Variables

In [12]:
width = 6
height = 8

n = width * height

print(n)

48


In [13]:
GST = 6/100
price = 88.50

price * GST

5.31

In [14]:
price + _

93.81

GST Calculation:
Given GST rate: 6% (which is equivalent to 0.06 in decimal form).
Price: $88.50.
GST amount = Price × GST rate = $88.50 × 0.06 = $5.31.
Total Price Including GST:
Total price = Price + GST amount = $88.50 + $5.31 = $93.81.
Now, let’s find the value of the placeholder (price + _):

Placeholder value = Total price including GST - Price = $93.81 - $88.50 = $5.31.
Therefore, the value of the placeholder (price + _) is $5.31.

# Self Practice

In [16]:
intial_investment = 100
return_on_investment = 1.1

After_7_year = 100 * (1.1 ** 7)

print(After_7_year)

194.87171000000012


4. Strings

In [17]:
"Hello"

'Hello'

In [19]:
"Hello"

'Hello'

No difference

In [20]:
'doesn\'t'

"doesn't"

In [21]:
"\"Yes, \" he said."

'"Yes, " he said.'

In [22]:
print('"Isn\'t, " she said.')

"Isn't, " she said.


In [23]:
s = 'First line. \nSecond line.'

In [24]:
s

'First line. \nSecond line.'

In [25]:
print(s)

First line. 
Second line.


In [26]:
print('C:\some\name')

C:\some
ame


In [27]:
print(r'C:\some\name')

C:\some\name


In [28]:
print("""\
    Ingredients:
                Plain flour 100g
                Milk 300 ml
                Caster suger (optional)
""")

    Ingredients:
                Plain flour 100g
                Milk 300 ml
                Caster suger (optional)



In [30]:
'ba' + 2 * 'na'

'banana'

In [32]:
header = 'ba'
body = 'nana'
header + body

'banana'

In [33]:
word = 'Minion'

word[0]

'M'

In [34]:
word[3]

'i'

In [35]:
word[-1]

'n'

In [36]:
word[-2]

'o'

In [37]:
word[-6]

'M'

counting from the end of the string

In [38]:
word[0:3]

'Min'

In [39]:
word[2:5]

'nio'

In [40]:
word[:3]

'Min'

In [41]:
word[4:]

'on'

In [42]:
word[:2] + word[2:]

'Minion'

In [44]:
word[-2:]

'on'

In [45]:
word[42]

IndexError: string index out of range

In [46]:
word[3:42]

'ion'

In [47]:
word[42:]

''

error and empty string

In [48]:
s = "DespicableMe"
len(s)

12